tensorflow/compiler/xla/service/hlo_graph_dumper.cc

```diff
@@ -1427,7 +1427,8 @@ NodeFilter MakeNodeFromToFilter(const HloInstruction* from,
     for (const auto* user : path.back()->users()) {
       if (user == to) {
         auto it = path.begin();
-        for (; it != path.end() && to_display.size() < max_nodes; ++it) {
+        const int64 to_display_size = to_display.size();
+        for (; it != path.end() && to_display_size < max_nodes; ++it) {
           to_display.insert(*it);
         }
```

Comment:
This is not good. You are inserting elements into `to_display`.

No warning on line 1420? If there are none there, there should be none here either, since it is the same comparison.